client/src/app/auth/services/auth.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { tap } from 'rxjs';
+import { UserModel } from '../auth.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  user: UserModel = {
+    email: '',
+    id: '',
+    token: '',
+  };
+
+  constructor(private http: HttpClient) {}
+
+  login(email: string, password: string) {
+    return this.http
+      .post('http://localhost:5000/api/user/login', {
+        email,
+        password,
+      })
+      .pipe(
+        tap((res: any) => {
+          this.setToken(res);
+          this.user.email = res.email;
```

Comment:
Why don't just this.user = res ?

Reply:
i did not use that because objects are reference types but in this case maybe  it won't be really a problem